### PR TITLE
feat: sticky headers

### LINKED ISSUE #25: Sticky Headers
- [x] Toolbar is fixed to the top
- [x] Headers stay in view when scrolling using the sticky property

### PR DESCRIPTION
**Pull request by rdsov**
Resolves #25

### CODE REVIEW

**Comment by rdsov** [comment]
Screenshot 2024-11-11 at 2 44 06 PM

File: src/plugin-store.svelte [hunk 1]
```diff
@@ -205,6 +207,7 @@ let pluginStore: Element;
     :root,
     :host {
         --header-height: 128px;
+        scrollbar-color: var(--color-grey-2) var(--mdc-theme-surface);
```
Comment (rdsov) [inline src/plugin-store.svelte:210]: Changed scrollbar background color to be consistent with off-white tinted background.

File: src/plugin-store.svelte [hunk 2]
```diff
@@ -110,18 +110,17 @@ let pluginStore: Element;
...
-        <Header>
-            <Title id="plugin-store-title">Plugin Store</Title>
-            <IconButton class="material-icons" action="close">
-                <IconClose />
-            </IconButton>
-        </Header>
-        <Content id="plugin-store-content">
-            <plugin-store bind:this={pluginStore}>
-                <plugin-store-toolbar>
+        <Header style="flex-direction: column; box-shadow: none;">
```
Comment (trusz) [inline src/plugin-store.svelte:116]: is there a reason why you use style instead of a class?

File: src/plugin-store.svelte [hunk 3]
```diff
@@ -110,18 +110,17 @@ let pluginStore: Element;
...
+            <plugin-store-header--top>
+                <Title id="plugin-store-header--title">Plugin Store</Title>
```
Comment (trusz) [inline src/plugin-store.svelte:118]: what purpose have the `ids`?

Comment (rdsov) [reply src/plugin-store.svelte:118]: It's used by MUI's Dialog attributes: `aria-labelledby` and `aria-describedby`.

File: src/plugin-store.svelte [hunk 4]
```diff
@@ -135,9 +134,12 @@ let pluginStore: Element;
                         bind:value={searchFilter}
                     />
                 </plugin-store-toolbar>
+        </Header>
+        <Content id="plugin-store-content">
+            <plugin-store bind:this={pluginStore}>
                 <plugin-store-items>
-                    <plugin-store-items--category>
-                        <strong><div class="mdc-typography--headline6">Editor</div></strong>
+                    <plugin-store-items--category style="margin-top: 0;">
```
Comment (trusz) [inline src/plugin-store.svelte:141]: one more style left :)

**Review by trusz — APPROVED** [review]

**Pull request merged**